omniscidb/QueryEngine/CostModel/CostModel.cpp

```diff
@@ -49,15 +53,21 @@ void CostModel::calibrate(const CaibrationConfig& conf) {
 
     for (auto& templateMeasurement : dmEntry.second) {
       AnalyticalTemplate templ = templateMeasurement.first;
+
+#ifdef HAVE_ARMADILLO
+      dp_[device][templ] =
+          std::make_unique<LinearRegression>(std::move(templateMeasurement.second));
+#else
```

Comment:
Looks like this can be resolved by inheritance instead of macros.

Reply:
I made a provider for extrapolation, but still left defines. just took them out into it. maybe I misunderstood this comment anyway?